### PR TITLE
feat(query): implement st_makeline() function

### PR DESCRIPTION
**Pull request by kkk25641463**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Support  snowflake st_makeline(geometry, geometry) function.

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

File: src/query/functions/src/scalars/geometry.rs [hunk 1]
```diff
@@ -67,6 +74,81 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+        vectorize_with_builder_2_arg::<GeometryType, GeometryType, GeometryType>(
+            |left_exp, right_exp, builder, ctx| {
+                if let Some(validity) = &ctx.validity {
+                    if !validity.get_bit(builder.len()) {
+                        builder.commit_row();
+                        return;
+                    }
+                }
+
+                let geometries: Vec<Geometry> = match binary_to_geometry(&vec![left_exp, right_exp])
```
Comment (ariesdevil) [inline src/query/functions/src/scalars/geometry.rs:89]: Seems we need impl an `Into<Geometry>` to simply convert.

File: src/query/functions/src/scalars/geometry.rs [hunk 2]
```diff
@@ -67,6 +74,81 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                {
+                    Ok(g) => g,
+                    Err(e) => {
+                        ctx.set_error(builder.len(), e.to_string());
+                        return builder.put_str("");
+                    }
+                };
+
```
Comment (ariesdevil) [inline src/query/functions/src/scalars/geometry.rs:97]: We need check the geometries have same SRID

File: src/query/functions/src/scalars/geometry.rs [hunk 3]
```diff
@@ -67,6 +74,81 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                        _ => {
+                            ctx.set_error(
+                                builder.len(),
+                                "Geometry expression must be a Point, MultiPoint, or LineString.",
+                            );
+                            return builder.put_str("");
+                        }
+                    }
+                }
+                let geom = Geometry::from(LineString::new(coords));
```
Comment (ariesdevil) [inline src/query/functions/src/scalars/geometry.rs:142]: Also, make sure the returned value has the same SRID as the inputs

File: src/query/functions/src/scalars/geometry.rs (hunk 1 above)
Comment (kkk25641463) [reply src/query/functions/src/scalars/geometry.rs:89]: I I simulate the implementation of PostGIS [code] (https://github.com/postgis/postgis/blob/88c3893c718847867714dc1bfddfa5db7c0a55e3/liblwgeom/lwline.c#L182C1-L203C4); The [functions](https://postgis.net/docs/ST_MakeLine.html) define in postgis are more flexible than snowflake;
To Implement Into<Geometry> function , we must include geozero crate in georust/geo project to process wkb logic, I'm not sure it's accepted by the community.

Comment (ariesdevil) [reply src/query/functions/src/scalars/geometry.rs:89]: Seems impl [this trait](https://docs.rs/geozero/latest/geozero/trait.ToGeos.html) is enough.

Comment (ariesdevil) [reply src/query/functions/src/scalars/geometry.rs:89]: Although postgis maybe more flexible, we should follow the Snowflake way, Databend is a Snowflake alternative.

**Comment by kkk25641463** [comment]
I had already add srid logic and unit test.

**Comment by ariesdevil** [comment]
> I had already add srid logic and unit test.

Plz run `make lint` make the ci happy

**Comment by kkk25641463** [comment]
> > I had already add srid logic and unit test.
> 
> Plz run `make lint` make the ci happy
Sorry, I'm not familiar with databend pr workflow.I'll fix it right now.

File: src/query/functions/src/scalars/geometry.rs (hunk 1 above)
Comment (kkk25641463) [reply src/query/functions/src/scalars/geometry.rs]: We need commit in geozero for trait ToGeos further.

**Review by ariesdevil — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**